sdks/python/apache_beam/io/gcp/pubsub.py

```diff
@@ -444,3 +448,123 @@ def display_data(self):
 
   def writer(self):
     raise NotImplementedError
+
+
+class MultipleReadFromPubSub(PTransform):
+  """A ``PTransform`` that expands ``ReadFromPubSub`` to read from multiple
+  subscriptions and/or topics."""
+  def __init__(
+      self,
+      source_list,  # type: List[str]
+      with_context=False,  # type: bool
+      id_label=None,  # type: Optional[Union[List[str], str]]
+      with_attributes=False,  # type: bool
+      timestamp_attribute=None,  # type: Optional[Union[List[str], str]]
+      **kwargs
+  ):
+    """Initializes ``PubSubMultipleReader``.
+
+    Args:
+      source_list: List of Cloud Pub/Sub topics or subscriptions. Topics in
+        form "projects/<project>/topics/<topic>" and subscriptions in form
+        "projects/<project>/subscriptions/<subscription>".
+      with_context:
+        True - output elements will be key-value pairs with the source as the
+        key and the message as the value.
+        False - output elements will be the messages.
+      with_attributes:
+        True - input elements will be :class:`~PubsubMessage` objects.
+        False - input elements will be of type ``bytes`` (message data only).
+      id_label: If set, will set an attribute for each Cloud Pub/Sub message
+        with the given name and a unique value. This attribute can then be
+        used in a ReadFromPubSub PTransform to deduplicate messages. If type
+        is string, all sources will share the same value; if type is
+        ``List``, each source will use the value of its index.
+      timestamp_attribute: If set, will set an attribute for each Cloud
+        Pub/Sub message with the given name and the message's publish time as
+        the value. If type is ``string``, all sources will share the same
+        value; if type List, each source will use the value of its index.
+    """
+    self.source_list = source_list
+    self.with_context = with_context
+    self.with_attributes = with_attributes
+    self._kwargs = kwargs
+
+    self._total_sources = len(source_list)
+
+    if isinstance(id_label, str) or id_label is None:
```

Comment:
Lots of logic here is checking input arguments, which we can get rid of by using a `NamedTuple` to represent a reading PubSub configuration instead of using a list for each configurable attributes.